From c1aa71ae1dee02bae491ccc72cd63fe75e964ade Mon Sep 17 00:00:00 2001
From: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Date: Fri, 21 Jun 2024 16:33:13 +0800
Subject: [PATCH 148/150] Revert "xHCI: Add XHCI_SLOWDOWN_QUIRK quirk for
 phytium xHCI controller"

This is a patch that is independent of the kernel code
and should not be merged.

Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Signed-off-by: Li Mingzhe <limingzhe1839@phytium.com.cn>
This reverts commit f32e26514729a2930eff87ca15822821b50aa7d2.
Change-Id: Ie67b36a08ee9c9a0af0a5bc853e068b835ded3db
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 drivers/usb/host/xhci-mem.c | 4 ----
 drivers/usb/host/xhci-pci.c | 4 +---
 drivers/usb/host/xhci.h     | 1 -
 3 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/usb/host/xhci-mem.c b/drivers/usb/host/xhci-mem.c
index 38b631cacd97..54c47463c215 100644
--- a/drivers/usb/host/xhci-mem.c
+++ b/drivers/usb/host/xhci-mem.c
@@ -1998,10 +1998,6 @@ static void xhci_add_in_port(struct xhci_hcd *xhci, unsigned int num_ports,
 		return;
 	}
 
-	if (xhci->quirks & XHCI_PHYTIUM_SLOWDOWN_QUIRK)
-		if (major_revision == 0x03)
-			rhub->min_rev = 0;
-
 	/* Port offset and count in the third dword, see section 7.2 */
 	temp = readl(addr + 2);
 	port_offset = XHCI_EXT_PORT_OFF(temp);
diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index 2eeac75dbd4d..6f08b76c9622 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -487,10 +487,8 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
 		xhci->quirks |= XHCI_RESET_ON_RESUME;
 
 	if (pdev->vendor == PCI_VENDOR_ID_PHYTIUM ||
-	    pdev->device == PCI_DEVICE_ID_PHYTIUM_XHCI) {
+	    pdev->device == PCI_DEVICE_ID_PHYTIUM_XHCI)
 		xhci->quirks |= XHCI_RESET_ON_RESUME;
-		xhci->quirks |= XHCI_PHYTIUM_SLOWDOWN_QUIRK;
-	}
 
 	/* See https://bugzilla.kernel.org/show_bug.cgi?id=79511 */
 	if (pdev->vendor == PCI_VENDOR_ID_VIA &&
diff --git a/drivers/usb/host/xhci.h b/drivers/usb/host/xhci.h
index 2e94376d4de2..f2190d121233 100644
--- a/drivers/usb/host/xhci.h
+++ b/drivers/usb/host/xhci.h
@@ -1660,7 +1660,6 @@ struct xhci_hcd {
 #define XHCI_ZHAOXIN_HOST	BIT_ULL(46)
 #define XHCI_WRITE_64_HI_LO	BIT_ULL(47)
 #define XHCI_CDNS_SCTX_QUIRK	BIT_ULL(48)
-#define XHCI_PHYTIUM_SLOWDOWN_QUIRK	BIT_ULL(49)
 
 	unsigned int		num_active_eps;
 	unsigned int		limit_active_eps;
-- 
2.47.0

